### DISCUSSION TITLE
Electric for Node/Deno/Bun?

### DISCUSSION THREAD

**Discussion by canadaduane** [hyperfiddle/electric#75] [General]
I'm a web developer, a Sr. Software engineer at an AI edtech company.

I admit I was absolutely blown away by Electric when it was first presented to the news.ycombinator.com community several years ago. "This looks utterly amazing!" I thought to myself, and I looked to see what language or platform it was available for. "Oh. Clojure. I'll wait until it's mainstream," I thought.

I also admit that was a bit of a snobbish thought. But the ecosystems I work in do not overlap very much with Clojure. There's the JVM hurdle (my gut reaction: old, hard to work with, isolated) and then there's the lisp hurdle (my gut reaction: cool tech, but I can't convince a modern web stack team to build on it). Finally, there's just the practical matter of React and Next.js dominating everything these days.

I really want to use Electric. But there are so many barriers to entry. I would love to hear about experiments in making it more compatible with and accessible to people like me who swim in different ponds.

**Comment by astrashe** [comment]
I think that Electric depends on Clojure's support for macros, and that it would be hard to implement in a language that doesn't have something similar.

**Reply by canadaduane** [reply]
Yes, that makes sense. I do enjoy learning powerful languages (I loved building with and learning Haskell, for example). One way that others have escaped the constraints of the target ecosystem is to build a language "on top". For example: [Imba](https://imba.io/). It has a number of excellent affordances that cannot be accomplished in Typescript alone, but it compiles to javascript and makes its mark on the ecosystem.
